### PR TITLE
Remove usage of `fbjs` from most packages

### PR DESCRIPTION
**Pull request by cruzach**
# Why

We rely on an old version of `fbjs` that has a vulnerability (reported by a partner), and in the case of all of these packages we only use it for `canUseDOM` (and also `invariant` in the expo-linking package). They also state in the `fbjs` readme:
> Note: If you are consuming the code here and you are not also a Facebook project, be prepared for a bad time.

so probably better we don't rely on it if we don't need to

Note: at the time of this PR `fbjs` still remains in `reanimated` (vendored) and `expo-screen-orientation` ( which requires changes to `@unimodules` packages, so it's better to do it in a [separate PR](https://github.com/expo/expo/pull/11398))

# How

Replaced instances of `canUseDOM` from fbjs with `Platform.isDomAvailable from '@unimodules/core'` 

Replaced instances of `invariant` from fbjs `invariant from 'invariant'`

# Test Plan

tests pass

### CODE REVIEW

File: packages/expo-constants/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 ### 🐛 Bug fixes
 
+- Removed `fbjs`dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```
Comment (EvanBacon) [inline packages/expo-constants/CHANGELOG.md:13]:
```suggestion
- Removed `fbjs` dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```

File: packages/expo-camera/CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,8 @@
 
 ### 🐛 Bug fixes
 
+- Removed `fbjs`dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```
Comment (cruzach) [inline packages/expo-camera/CHANGELOG.md:17]:
```suggestion
- Removed `fbjs` dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```

File: packages/expo-device/CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,8 @@
 
 ### 🐛 Bug fixes
 
+- Removed `fbjs`dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```
Comment (cruzach) [inline packages/expo-device/CHANGELOG.md:13]:
```suggestion
- Removed `fbjs` dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```

File: packages/expo-error-recovery/CHANGELOG.md [hunk 4]
```diff
@@ -10,6 +10,8 @@
 
 ### 🐛 Bug fixes
 
+- Removed `fbjs`dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```
Comment (cruzach) [inline packages/expo-error-recovery/CHANGELOG.md:13]:
```suggestion
- Removed `fbjs` dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```

File: packages/expo-font/CHANGELOG.md [hunk 5]
```diff
@@ -11,6 +11,7 @@
 ### 🐛 Bug fixes
 
 - Remove Expo.AppLoading reference in error. ([#11204](https://github.com/expo/expo/pull/11204) by [@brentvatne](https://github.com/brentvatne))
+- Removed `fbjs`dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```
Comment (cruzach) [inline packages/expo-font/CHANGELOG.md:14]:
```suggestion
- Removed `fbjs` dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```

File: packages/expo-gl/CHANGELOG.md [hunk 6]
```diff
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Removed `fbjs`dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```
Comment (cruzach) [inline packages/expo-gl/CHANGELOG.md:11]:
```suggestion
- Removed `fbjs` dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```

File: packages/expo-linking/CHANGELOG.md [hunk 7]
```diff
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Removed `fbjs`dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```
Comment (cruzach) [inline packages/expo-linking/CHANGELOG.md:11]:
```suggestion
- Removed `fbjs` dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```

File: packages/expo-notifications/CHANGELOG.md [hunk 8]
```diff
@@ -17,6 +17,7 @@
 
 - Fixed migration process to **not** use `expo-constants` installation ID if there is a notifications-specific identifier. ([#11287](https://github.com/expo/expo/pull/11287) by [@sjchmiela](https://github.com/sjchmiela))
 - Changed the visibility of Android's `InstallationId#getNonBackedUpUuidFile` method so it's easier to override by custom implementations. ([#11249](https://github.com/expo/expo/pull/11249) by [@sjchmiela](https://github.com/sjchmiela))
+- Removed `fbjs`dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```
Comment (cruzach) [inline packages/expo-notifications/CHANGELOG.md:22]:
```suggestion
- Removed `fbjs` dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```

File: packages/expo-sensors/CHANGELOG.md [hunk 9]
```diff
@@ -14,6 +14,8 @@
 
 ### 🐛 Bug fixes
 
+- Removed `fbjs`dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```
Comment (cruzach) [inline packages/expo-sensors/CHANGELOG.md:17]:
```suggestion
- Removed `fbjs` dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```

File: packages/expo-updates/CHANGELOG.md [hunk 10]
```diff
@@ -16,6 +16,8 @@
 
 ### 🐛 Bug fixes
 
+- Removed `fbjs`dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```
Comment (cruzach) [inline packages/expo-updates/CHANGELOG.md:19]:
```suggestion
- Removed `fbjs` dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```

File: packages/expo-web-browser/CHANGELOG.md [hunk 11]
```diff
@@ -10,6 +10,8 @@
 
 ### 🐛 Bug fixes
 
+- Removed `fbjs`dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```
Comment (cruzach) [inline packages/expo-web-browser/CHANGELOG.md:13]:
```suggestion
- Removed `fbjs` dependency ([#11396](https://github.com/expo/expo/pull/11396) by [@cruzach](https://github.com/cruzach))
```

**Review by ide — APPROVED** [review]
As a follow-up commit, I'd suggest modifying `expo-module-scripts/eslintrc.base.js` and adding something like:
```js
"no-restricted-imports": ["warn", {
    "paths": [{ name: "fbjs", message: "fbjs is a Facebook-internal package not intended to be a public API" }]
}]
```

(test if this matches importing from `fbjs/stuff` too)

**Review by tsapeta — APPROVED** [review]
lgtm! 👍

**Review by lukmccall — APPROVED** [review]

**Review by bbarthec — APPROVED** [review]

**Pull request merged**